src/ripple/app/paths/Pathfinder.cpp

```diff
@@ -1185,8 +1185,10 @@ void fillPaths (Pathfinder::PaymentType type, PathCostList const& costs)
 void Pathfinder::initPathTable ()
 {
     // CAUTION: Do not include rules that build default paths
-    fillPaths(
-        pt_XRP_to_XRP, {});
+
+    // When the app is destroyed and a new app is created, start with a fresh mPathTable
+    mPathTable.clear();
+    fillPaths (pt_XRP_to_XRP, {});
```

Comment:
Since mPathTable is internal to Pathfinder.cpp I thought it the comment was better in the cpp file.